src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -229,10 +237,25 @@ def _compute_metrics(self, sorting_analyzer, unit_ids=None, verbose=False, metri
         # add NaN for empty units
         if len(empty_unit_ids) > 0:
             metrics.loc[empty_unit_ids] = np.nan
+            # num_spikes is an int and should be 0
+            if "num_spikes" in metrics.columns:
+                metrics.loc[empty_unit_ids, ["num_spikes"]] = 0
 
         # we use the convert_dtypes to convert the columns to the most appropriate dtype and avoid object columns
         # (in case of NaN values)
         metrics = metrics.convert_dtypes()
+
+        # we do this because the convert_dtypes infers the wrong types sometimes.
+        # the actual types for columns can be found in column_name_to_column_dtype dictionary.
+        for column in metrics.columns:
+            # we have one issue where the name of the columns for synchrony are named based on
+            # what the user has input as arguments so we need a way to handle this separately
+            # everything else should be handled with the column name.
+            if "sync" in column:
+                metrics[column] = metrics[column].astype(column_name_to_column_dtype["sync"])
+            else:
+                metrics[column] = metrics[column].astype(column_name_to_column_dtype[column])
```

Comment:
```suggestion
            metrics[column] = metrics[column].astype(column_name_to_column_dtype[column])
```
https://github.com/SpikeInterface/spikeinterface/pull/3559